src/extensions/codemirror/CodeMirror.tsx

```diff
@@ -98,108 +83,95 @@ export interface CodeEditorProps {
     /**
      *  handler for scroll event
      */
-    onScroll?: (editorInstance: CodeMirror.Editor) => void;
+    onScroll?: (event: Event, view: EditorView) => boolean | void;
     /**
      * optional property to fold code for the supported modes e.g: xml, json etc.
      */
     supportCodeFolding?: boolean;
+    /**
+     * highlight active line where the cursor is currently in
+     */
+    shouldHighlightActiveLine?: boolean;
 }
 
-/**
- * Includes a code editor, currently we use CodeMirror library as base.
- */
+const addExtensionsFor = (flag: boolean, ...extensions: Extension[]) => (flag ? [...extensions] : []);
+const addToKeyMapConfigFor = (flag: boolean, ...keys: any) => (flag ? [...keys] : []);
+const addHandlersFor = (flag: boolean, handlerName: string, handler: any) =>
+    flag ? ({ [handlerName]: handler } as DOMEventHandlers<any>) : {};
+
 export const CodeEditor = ({
     onChange,
-    name,
     id,
-    mode = "undefined",
+    mode,
     preventLineNumbers = false,
     defaultValue = "",
     readOnly = false,
-    height,
     wrapLines = false,
     onScroll,
-    setEditorInstance,
+    setEditorView,
     supportCodeFolding = false,
+    shouldHighlightActiveLine = false,
     outerDivAttributes,
     tabIntentSize = 2,
     tabIntentStyle = "tab",
     tabForceSpaceForModes = ["python", "yaml"],
+    height = 290,
```

Comment:
`height` is used as inline style later. Why is this not controlled via stylesheets?